src/node_report_utils.cc

```diff
@@ -0,0 +1,306 @@
+      if (rc == 0) {
+        data << "width: " << width << ", height: " << height;
+      }
+      break;
+    }
+    case UV_UDP: {
+      type = "udp";
+      ReportEndpoints(h, data);
+      break;
+    }
+    case UV_SIGNAL: {
+      // SIGWINCH is used by libuv so always appears.
+      // See http://docs.libuv.org/en/v1.x/signal.html
+      type = "signal";
+      data << "signum: " << handle->signal.signum
+      // node::signo_string() is not exported by Node.js on Windows.
+#ifndef _WIN32
+           << " (" << node::signo_string(handle->signal.signum) << ")"
+#endif
+           << "";
+      break;
+    }
+    case UV_FILE: type = "file"; break;
+    // We shouldn't see "max" type
+    case UV_HANDLE_TYPE_MAX : type = "max"; break;
+  }
+
+  if (h->type == UV_TCP || h->type == UV_UDP
+#ifndef _WIN32
+      || h->type == UV_NAMED_PIPE
+#endif
+      ) {
+    // These *must* be 0 or libuv will set the buffer sizes to the non-zero
+    // values they contain.
+    int send_size = 0;
+    int recv_size = 0;
+    if (h->type == UV_TCP || h->type == UV_UDP) {
+      data << ", ";
+    }
+    uv_send_buffer_size(h, &send_size);
+    uv_recv_buffer_size(h, &recv_size);
+    data << "send buffer size: " << send_size
+         << ", recv buffer size: " << recv_size;
+  }
+
+  if (h->type == UV_TCP || h->type == UV_NAMED_PIPE || h->type == UV_TTY ||
+      h->type == UV_UDP || h->type == UV_POLL) {
+    uv_os_fd_t fd_v;
+    uv_os_fd_t* fd = &fd_v;
+    int rc  = uv_fileno(h, fd);
+    // uv_os_fd_t is an int on Unix and HANDLE on Windows.
+#ifndef _WIN32
+    if (rc == 0) {
+      switch (fd_v) {
+      case 0:
+        data << ", stdin"; break;
+      case 1:
+        data << ", stdout"; break;
+      case 2:
+        data << ", stderr"; break;
+      default:
+        data << ", file descriptor: " << static_cast<int>(fd_v);
+        break;
+      }
+    }
+#endif
+  }
+
+  if (h->type == UV_TCP || h->type == UV_NAMED_PIPE || h->type == UV_TTY) {
+    data << ", write queue size: "
+         << handle->stream.write_queue_size;
+    data << (uv_is_readable(&handle->stream) ? ", readable" : "")
+         << (uv_is_writable(&handle->stream) ? ", writable": "");
+  }
+
+  json_start(*out, *indent);
+  json_indent(indent);
+  json_keyvaluenormal(*out, *indent, "type", type);
+  json_keyvaluenormal(*out, *indent, "is_active", uv_is_active(h));
+  json_keyvaluenormal(*out, *indent, "is_referenced", uv_has_ref(h));
+  json_keyvaluenormal(*out, *indent, "address", static_cast<void*>(h));
+  json_keyvaluefinal(*out, *indent, "details", data.str());
+  json_deindent(indent);
+  json_objectendnormal(*out, *indent);
+}
+
+/*******************************************************************************
+ * Utility function to print out integer values with commas for readability.
+ ******************************************************************************/
+void WriteInteger(std::ostream& out, size_t value) {
```

Comment:
This function is currently unused, and tbh I’d be a fan of not doing formatting on the numbers :)

Reply:
removed the unused function